src/integration-test/search.test.ts

```diff
@@ -0,0 +1,46 @@
+import * as assert from 'assert'
+import { take } from 'rxjs/operators'
+import { integrationTestContext } from './helpers.test'
+
+describe('search (integration)', () => {
+    it('registers a query transformer', async () => {
+        const { clientController, extensionHost, ready } = await integrationTestContext()
+
+        // Register the provider and call it
+        const unsubscribe = extensionHost.search.registerQueryTransformer({ transformQuery: () => 'bar' })
+        await ready
+        assert.deepStrictEqual(
+            await clientController.registries.queryTransformer
+                .transformQuery('foo')
+                .pipe(take(1))
+                .toPromise(),
+            'bar'
+        )
+
+        // Unregister the provider and ensure it's removed
+        unsubscribe.unsubscribe()
+        assert.deepStrictEqual(
+            await clientController.registries.queryTransformer
+                .transformQuery('foo')
+                .pipe(take(1))
+                .toPromise(),
+            'foo'
+        )
+    })
+
+    it('supports multiple query transformers', async () => {
+        const { clientController, extensionHost, ready } = await integrationTestContext()
+
+        // Register the provider and call it
+        extensionHost.search.registerQueryTransformer({ transformQuery: q => `${q} bar` })
```

Comment:
Minor, but `bar` and `baz` look almost the same; it's not easy to visually spot that they are distinct, given that everything else on the 2 lines is the same. I would recommend using `foo`, `qux`, etc., here on one of the lines so it's easier to spot that they are different.